src/loaders/genome_collection/parse_tool_results.py

```diff
@@ -640,13 +640,28 @@ def _retrieve_sample(root_dir, env, kbase_collection, source_ver, load_ver):
     data_id_sample_id_map = {doc[names.FLD_KBASE_ID]: doc[names.FLD_KB_SAMPLE_ID] for doc in prepared_samples_data}
 
     flatten_samples_data = _flat_samples_data(prepared_samples_data)
+    flatten_samples_data, meta_doc = process_columnar_meta(
+        flatten_samples_data,
+        kbase_collection,
+        load_ver,
+        names.SAMPLES_PRODUCT_ID,
+        ignore_missing=True,
```

Comment:
Doesn't `process_columnar_meta` process based on the collection, so we shouldn't need this since it won't look for GROW keys in the ENIGMA collection and vice versa?

Reply:
It won't look for GROW keys in the ENIGMA collections and vice versa. Yes. It is intended to be used within the collection. Some docs within the collection are missing some of the defined keys in the spec file. It happens to both collections unfortunately.

Reply:
Are there specific samples that are missing the key or is the key missing from all samples?

Reply:
Some samples are missing keys. For example:
```
ValueError: Unable to find key: country in {'kbase_ids': ['69739_100_1', '69739_140_1', '69739_116_1', '69739_158_1', '69739_106_1', '69739_110_1', '69739_176_1', '69739_120_1', '69739_118_1', '69739_114_1', '69739_112_1', '69739_104_1', '69739_164_1'], '_key': '1def08d77168be50df480f5840aadc68', 'coll': 'ENIGMA', 'load_ver': '2023.11', '_mtchsel': [], 'enigma:experiment_name': 'Natural Organic Matter', 'enigma:well_name': 'FW-305', 'env_package': 'water', 'material': 'ENVO:01001004', 'enigma:date': '2016-03-31', 'enigma:time_zone': 'UTC−04:00', 'latitude': 35.94091048, 'longitude': -84.33614124999998, 'sample_template': 'ENIGMA', '_geo_spatial': [-84.33614124999998, 35.94091048], 'kbase_sample_id': 'c1b86110-6372-492c-9ced-d36393e7b674', 'kbase_display_name': 'FW305-033116', 'genome_count': 13}
```
Most of other samples do have `country` key.

Reply:
Here is another example from GROW:
```
ValueError: Unable to find key: env_package in {'kbase_ids': "['69037_2436_1', '69037_2480_1', '69037_2502_1', '69037_2496_1', '69037_2472_1', '69037_2498_1', '69037_2432_1', '69037_2484_1', '69037_2486_1', '69037_2494_1', '69037_2448_1', '69037_2490_1', '69037_2434_1', '69037_2452_1', '69037_2506_1', '69037_2508_1', '69037_2446_1', '69037_2468_1', '69037_2444_1', '69037_2500_1', '69037_2456_1', '69037_2466_1', '69037_2488_1', '69037_2482_1', '69037_2510_1', '69037_2438_1', '69037_2454_1', '69037_2504_1', '69037_2464_1', '69037_2442_1', '69037_2470_1', '69037_2478_1', '69037_2440_1', '69037_2460_1', '69037_2492_1', '69037_2450_1', '69037_2458_1', '69037_2476_1', '69037_2462_1', '69037_2474_1']", '_key': '9a612376a324096c3a4dbbbfba6f1a6d', 'coll': 'GROW', 'load_ver': '2023.11', '_mtchsel': '[]', 'sesar:igsn': 'IEGRW000F', 'sesar:material': 'Liquid>aqueous', 'sesar:field_name': 'surface water', 'other_names': 'tombigeeriver_2019_sw_WHONDRS-S19S_0020', 'sesar:collection_method': 'Grab', 'sesar:collection_method_description': 'surface water filtered through 0.2micron filter; see collection protocol published at https://data.ess-dive.lbl.gov/view/doi:10.15485/1729719', 'purpose': 'S19S sampling of global rivers for WHONDRS', 'latitude': 31.852940000000004, 'longitude': -88.15755, 'sesar:physiographic_feature_primary': 'stream', 'sesar:physiographic_feature_name': 'Tombigbee River', 'country': 'United States', 'sesar:field_program_cruise': 'WHONDRS', 'sesar:collector_chief_scientist': 'Jamie Galloway', 'sesar:collection_date': '2019-08-19', 'sesar:archive_contact_current': 'Mikayla Borton', 'sample_template': 'SESAR', '_geo_spatial': [-88.15755, 31.852940000000004], 'kbase_sample_id': 'e9d235ec-0c4e-464f-a776-da71f0f05901', 'kbase_display_name': 'S19S_0020', 'genome_count': 40}
```